src/compute-client/src/protocol/command.rs

```diff
@@ -513,6 +513,9 @@ pub struct Peek<T = mz_repr::Timestamp> {
     /// the compute controller and the compute worker.
     #[proptest(strategy = "empty_otel_ctx()")]
     pub otel_ctx: OpenTelemetryContext,
+    /// If this peek is for a Persist shard, the collection metadata for that shard.
+    /// (If absent, this is a standard peek against an arrangement.)
+    pub collection_meta: Option<CollectionMetadata>,
```

Comment:
An alternative to this would be two variants that have a common message. I think I'd prefer this over nested variants!

Reply:
Generally on board! I've added the `target` field. 

However - I would like to keep the `id: GlobalId` field for Persist-backed peeks as well. (Since an id exists and is useful in more or less the same cases.) I do appreciate that it's important not to introduce ambiguity in the logging... I'll do an audit there and follow up with any issues I run into.